docs/src/_reference/command-line-interface.md

```diff
@@ -732,11 +731,69 @@ meltano job remove <job_name>
 
 ### Examples
 
+##### Tasks and Blocks
+
+A task in a job consists of one or more _blocks_. A block is simply a set of related plugins that will be executed serially.
+For example, you can create a job that consists of a single tap and target:
+
+```bash
+meltano job add tap-gitlab-to-target-postgres --tasks "tap-gitlab target-postgres"
+```
+
+This would add the following to your `meltano.yml`:
+
+```yaml
+jobs:
+  - name: tap-gitlab-to-target-postgres
+    tasks:
+      - tap-gitlab target-postgres
+```
+
+Plugins in the same block should be separated by a space and blocks should be separted by a comma.
+
+You can also provide tasks in YAML format. For example, the following is equivalent to the above:
+
+```bash
+meltano job add tap-gitlab-to-target-postgres --tasks "[tap-gitlab target-postgres]"
+```
+
+Note that any single block which includes a tap _must_ also include a target and vice versa.
+The tap must also precede the target.
+
+All of the following are invalid:
+
+```bash
+# A block with a tap must also include a target.
+meltano job add invalid-job --tasks "tap-gitlab"
+>>> Job 'invalid-job' has invalid tasks. block violates set requirements: Found no end in block set!
+
+# Within a block, a target must be preceded by a tap.
+meltano job add invalid-job --tasks target-postgres
+>>> Job 'invalid-job' has invalid tasks. block violates set requirements: Unknown command type or bad block sequence at index 1, starting block 'target-postgres'
+
+# Blocks are separated by commas, so Meltano considers this tap and target as two separate invalid blocks.
+meltano job add invalid-job --tasks "tap-gitlab, target-postgres"
+>>> Job 'invalid-job' has invalid tasks. Block tap-gitlab, not found
+```
+
+A job can consist of multiple blocks, which will be executed serially in the order they are provided:
+
+```bash
+# This job consists of two blocks:
+meltano add tap-gitlab-to-target-postgres-with-dbt --tasks "[tap-gitlab target-postgres, dbt:run]"
+
+# This job consists of one block, but runs the same three tasks in the same order as above. It is functionally equivalent to the comma-sparated approach.
+meltano add tap-gitlab-to-target-postgres-with-dbt-single-block --tasks "tap-gitlab target-postgres dbt:run"
```

Comment:
Does this actually form one "tap-gitlab target-postgres dbt:run" block as the comment suggests? I was under the impression that only "tap [transformer?] target" combinations form a block together, and our `meltano run` argument parsing logic would also interpret `"tap-gitlab target-postgres dbt:run"` as 2 blocks.

Reply:
Yeah I suppose I've overloaded the term "blocks" here when just sticking to "task" would mostly do the job. Under the hood, `"tap-gitlab target-postgres dbt:run"` definitely gets parsed into one `ExtractLoadBlock` for the tap+target and then one `InvokerCommand` for the `dbt:run`. But it gets written to a single array element in the `meltano.yml`, which is what I was considering synonymous with a "block" here, whereas maybe "task" is the better term for that.